### ISSUE TITLE
Profile not applied automatically since version 3.9.3-1

### ISSUE THREAD

**Issue by sushi2503** [eoyilmaz/displaycal-py3#113]
Hi,

Since version 3.9.3-1 (yesterday's update) the profile won't load automatically and I have to install it again at every new session and boot. Even the displaycal-apply-profiles --force command does nothing.

Archlinux up to date.

This is the content of DisplayCAL.log

```shell
2022-05-16 10:13:03,808 ================================================================================
2022-05-16 10:14:34,641 /usr/lib/python3.10/site-packages/DisplayCAL/colord.py:182: Warning: CDError("gi.repository.GLib.Error: GLib.Error('GDBus.Error:org.freedesktop.ColorManager.NotFound: property match 'OutputEdidMd5'='46166769ae2ad938673e1d533b8cdb5e' does not exist', 'g-io-error-quark', 36)")
2022-05-16 10:14:34,641   warnings.warn(safe_str(exception), Warning)
```

Then the terminal logs by opening Displaycal and installing the profil : 

```shell
displaycal
Acquired lock file: <DisplayCAL.main.AppLock object at 0x7fd6ca936290>
displaycal 3.9.3 2022-05-04T20:16:01Z
arch   x86_64
Python 3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0]
Faulthandler 
wxPython 4.0.7.post2 gtk3 (phoenix) wxWidgets 3.0.5
Encoding: utf-8
File system encoding: utf-8
Loading /home/pilote/.config/DisplayCAL/DisplayCAL.ini
Existing client using port 15411
Connecting to 15411...
Connection to 127.0.0.1:15411 failed: [Errno 111] Connection refused
listening
writing to lock file: port: 34703
Démarrage…
SDL2: libSDL2-2.0.so.0
Module audio : wx 4.0.7.post2
Inventaire des périphériques d’affichage et des ports de communication…
/usr/bin
ArgyllCMS 2.3.0
Argyll has virtual display support
… opération terminée.
Contrôle de l’accès à la table de gamma de la carte vidéo pour l’écran 1…
Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10
VideoLUT has 1024 entries, interpolating to 256
Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10
Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10
Verify: 'test.cal' IS loaded (discrepancy 0.0%)
Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10
… opération terminée.
Initialisation de l’interface graphique utilisateur/trice…

… opération terminée.
Prêt.
Configuration de l’hôte de scripting à 127.0.0.1:34703
Vérifier si l’application est à jour…
DisplayCAL est à jour.
ArgyllCMS est à jour.
Found display device matching model description at index #0
Chargement des courbes d’étalonnage à partir du fichier…
/home/pilote/.local/share/DisplayCAL/storage/U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX/U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc
Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10
About to open dispwin object on the display
About to clear the calibration
About to set display to given calibration
Calibration set
About to destroy dispwin object
… opération terminée.
Authentification
--------------------------------------------------------------------------------
Installer le profil
--------------------------------------------------------------------------------
Session log: U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX

Répertoire de travail :
  /
   tmp/
    DisplayCAL-qk9ykpv3/

Ligne de commandes :
  /usr/bin/dispwin
    -v
    -d1
    -c
    -I
    'U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc'

About to open dispwin object on the display
About to clear the calibration
About to install 'U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc' as display's default profile
Installed 'U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc' and made it the default
About to destroy dispwin object
Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10
dispwin exitcode: 0
--------------------------------------------------------------------------------
Session log: U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX

Répertoire de travail :
  /
   tmp/
    DisplayCAL-qk9ykpv3/

Ligne de commandes :
  /usr/bin/dispwin
    -Sl
    -v
    -d1
    -c
    -I
    'U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc'

DisplayCAL: Waiting for EOF
Password:

About to open dispwin object on the display

Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10

About to clear the calibration

About to install 'U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc' as display's default profile

Installed 'U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc' and made it the default

About to destroy dispwin object

DisplayCAL: Reached EOF (OK)

dispwin exitcode: 0
/usr/lib/python3.10/site-packages/DisplayCAL/colord.py:182: Warning: CDError("gi.repository.GLib.Error: GLib.Error('GDBus.Error:org.freedesktop.ColorManager.NotFound: property match 'OutputEdidMd5'='46166769ae2ad938673e1d533b8cdb5e' does not exist', 'g-io-error-quark', 36)")
  warnings.warn(safe_str(exception), Warning)
DisplayCAL: Trying device ID 'xrandr-Dell, Inc.-DELL U2713HM-GK0KD31S262L'
About to overwrite existing /home/pilote/.local/share/icc/U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc
Writing /home/pilote/.local/share/icc/U2713HM #1 2021-12-18 17-39 120cdm² D6500 2.2 F-S XYZLUT+MTX.icc
CDError("gi.repository.GLib.Error: GLib.Error('GDBus.Error:org.freedesktop.ColorManager.NotFound: profile id 'icc-da206a6cceb31ba280651c8dff42c345' does not exist', 'g-io-error-quark', 36)")
--------------------------------------------------------------------------------
Répertoire de travail :
  /
   etc/
    xdg/
     DisplayCAL/

Ligne de commandes :
  mv
    -f
    /home/pilote/.config/DisplayCAL/DisplayCAL.local.ini
    DisplayCAL.ini

DisplayCAL: Waiting for EOF
Password:

DisplayCAL: Reached EOF (OK)
mv exitcode: 0
--------------------------------------------------------------------------------
Répertoire de travail :
  /
   etc/
    xdg/

Ligne de commandes :
  mkdir
    -p
    autostart

DisplayCAL: Waiting for EOF
Password:

DisplayCAL: Reached EOF (OK)
mkdir exitcode: 0
--------------------------------------------------------------------------------
Répertoire de travail :
  /
   etc/
    xdg/
     autostart/

Ligne de commandes :
  mv
    -f
    /home/pilote/.config/autostart/z-displaycal-apply-profiles.desktop
    z-displaycal-apply-profiles.desktop

DisplayCAL: Waiting for EOF
Password:

DisplayCAL: Reached EOF (OK)
mv exitcode: 0
┌──────────────────────────────────────────────────────────────────────────────┐
│ Le profil a été installé mais il peut y avoir des problèmes.                 │
└──────────────────────────────────────────────────────────────────────────────┘
NoneType: None
┌──────────────────────────────────────────────────────────────────────────────┐
│ Traceback (most recent call last):                                           │
│   File                                                                       │
│ "/usr/lib/python3.10/site-packages/DisplayCAL/lib/agw/fourwaysplitter.py",   │
│ line 774, in OnPaint                                                         │
│     self.DrawSplitter(dc)                                                    │
│   File "/usr/lib/python3.10/site-packages/DisplayCAL/wxwindows.py", line     │
│ 7574, in DrawSplitter                                                        │
│     dc.DrawRectangle(splitx, 0, sashwidth, height)                           │
│ TypeError: DC.DrawRectangle(): arguments did not match any overloaded call:  │
│   overload 1: argument 1 has unexpected type 'float'                         │
│   overload 2: argument 1 has unexpected type 'float'                         │
│   overload 3: argument 1 has unexpected type 'float'                         │
└──────────────────────────────────────────────────────────────────────────────┘
NoneType: None
┌──────────────────────────────────────────────────────────────────────────────┐
│ Traceback (most recent call last):                                           │
│   File                                                                       │
│ "/usr/lib/python3.10/site-packages/DisplayCAL/lib/agw/fourwaysplitter.py",   │
│ line 774, in OnPaint                                                         │
│     self.DrawSplitter(dc)                                                    │
│   File "/usr/lib/python3.10/site-packages/DisplayCAL/wxwindows.py", line     │
│ 7574, in DrawSplitter                                                        │
│     dc.DrawRectangle(splitx, 0, sashwidth, height)                           │
│ TypeError: DC.DrawRectangle(): arguments did not match any overloaded call:  │
│   overload 1: argument 1 has unexpected type 'float'                         │
│   overload 2: argument 1 has unexpected type 'float'                         │
│   overload 3: argument 1 has unexpected type 'float'                         │
└──────────────────────────────────────────────────────────────────────────────┘
Processus arrêté
```

Thanks for any help.

**Comment by eoyilmaz** [comment]
I don't know what ``3.9.3-1`` means by the way? I consider it like ``3.9.3``.

**Comment by sushi2503** [comment]
3.9.3-1 = 3.9.3 Yes, probably I kind of Archlinux way to indicate a version (?).

Where can the develop branch be found ? Is this issue also fixed or only the #114 ?

Thanks.

**Comment by eoyilmaz** [comment]
I fixed a couple more issues related to the ``FourWaySplitter``, so the splitter in the Profile Info window is working properly now.

To switch to the ``develop`` branch use:

```shell
git clone https://www.github.com/eoyilmaz/displaycal-py3
cd displaycal-py3
git checkout develop
```

**Comment by sushi2503** [comment]
I found the reason why the profile is not applied by seeing this in my xsession-errors : 

xfce4-session-WARNING **: 07:05:21.006: Unable to launch "b'displaycal-apply-profiles'" (specified by autostart/z-displaycal-apply-profiles.desktop):

Then I had a look at /etc/xdg/autostart/z-displaycal-apply-profiles.desktop and I found the error at the two lines : 

Icon="b'displaycal-apply-profiles'"
Exec="b'displaycal-apply-profiles'"

Instead of : 

Icon=displaycal-apply-profiles
Exec=displaycal-apply-profiles

Changing this makes the profile applying without problems and confirm by my xsession-errors content : 

Loading /home/pilote/.config/DisplayCAL/DisplayCAL.ini
================================================================================
Chargement des courbes d’étalonnage du profil actuel du périphérique d’affichage…
Inventaire des périphériques d’affichage et des ports de communication…
/usr/bin
ArgyllCMS 2.3.0
Argyll has virtual display support
… opération terminée.
decoding to str: need a bytes-like object, CDError found
--------------------------------------------------------------------------------
Ligne de commandes :
  /usr/bin/dispwin
    -v
    -d1
    -L

About to open dispwin object on the display
About to set display to given calibration
Calibration set
About to destroy dispwin object
Dispwin: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10
dispwin exitcode: 0

But there is still something wrong because the content in xsession-errors should appear in /home/USER/.local/share/DisplayCAL/logs (and not in xsession-errors) which looks like this : 

2022-05-17 07:05:39,066 /usr/lib/python3.10/site-packages/DisplayCAL/colord.py:182: Warning: CDError("gi.repository.GLib.Error: GLib.Error('GDBus.Error:org.freedesktop.ColorManager.NotFound: property match 'OutputEdidMd5'='46166769ae2ad938673e1d533b8cdb5e' does not exist', 'g-io-error-quark', 36)")
2022-05-17 07:05:39,066   warnings.warn(safe_str(exception), Warning)
2022-05-17 07:24:54,337 /usr/lib/python3.10/site-packages/DisplayCAL/colord.py:182: Warning: CDError("gi.repository.GLib.Error: GLib.Error('GDBus.Error:org.freedesktop.ColorManager.NotFound: property match 'OutputEdidMd5'='46166769ae2ad938673e1d533b8cdb5e' does not exist', 'g-io-error-quark', 36)")
2022-05-17 07:24:54,337   warnings.warn(safe_str(exception), Warning)

**Comment by eoyilmaz** [comment]
Cool, that's certainly a bug, I'll fix it later today :+1:

**Comment by eoyilmaz** [comment]
This should be fixed in the latest commit in the ``develop`` branch and with DisplayCAL 3.9.5.

**Comment by sushi2503** [comment]
Ok, thank you very much.

I will try the develop branch as soon as I can.

**Comment by eoyilmaz** [comment]
Ah now I see, I didn't know somebody was packaging DisplayCAL for Arch Linux. I see that you are using Arch's own package manager to install DisplayCAL. The packager is also reporting some problems with 3.9.4 (#117). But I believe it is not a valid problem. Anyway, your problem is solved, but you need to wait for 3.9.5 to be released for Arch.

**Comment by sushi2503** [comment]
Right, it's mentioned in my first message "Archlinux up to date". I could update to the 3.9.4 without problem.

Anyway, thanks for your precious work.

P.S. As I understand, Florian is about to live Displaycal, I didn't know.
